### PR TITLE
Improve lang-web.el to support zencoding, company-mode completions

### PR DESCRIPTION
**Pull request by CSRaghunandan**
Now there will be `company` completions in HTML/CSS and `<script>...</script>` 

Other changes:
* removed all the `:ensure t` statements since they are not necessary
* refactored code in lang-web.el to use `:mode`
* added [`impatitent-mode`](https://github.com/skeeto/impatient-mode) which provides live refreshing of web pages

### CODE REVIEW

**Comment by rakanalh** [comment]
Nice! Could you please rebase your branch on the latest master? i'll have a deeper look at this tonight.
Thanks.

**Comment by CSRaghunandan** [comment]
This is the first time I'm contributing a repository in github and honestly, I do not know how to rebase my fork to the latest master ( I've only done very basic stuff in git so far)
I tried rebasing in `magit`, I selected `origin/master` and nothing happened >.>
If you use magit, can you please tell me how I can rebase to master? If not, what git commands I need to execute?

**Comment by rakanalh** [comment]
Ok so basically you forked the repo right? so follow those two links in order:
https://help.github.com/articles/configuring-a-remote-for-a-fork/
https://help.github.com/articles/syncing-a-fork/

**Comment by CSRaghunandan** [comment]
@rakanalh  Done. Please check now.
btw, do you not use `magit` for your git workflow?

**Comment by rakanalh** [comment]
I do, but it was easier giving you links to the CLI version :)

File: README.org [hunk 1]
```diff
@@ -103,7 +107,20 @@ I would love to extend Emacs-bootstrap to support other programming languages as
 If you think you've got your configuration of your favorite programming language perfectly,
 please don't hesitate to submit a PR.
 
+<<<<<<< HEAD
```
Comment (rakanalh) [inline README.org:110]: See? This means a conflict is there and you should have fixed it.
`magit-ediff-resolve` in Emacs could have helped you resolve this. But i'll fix it this time.

File: emacs_templates/emacs/elisp/lang-web.el [hunk 2]
```diff
@@ -1,32 +1,94 @@
...
+	(let ((web-mode-cur-language
+	       (web-mode-language-at-pos)))
+	  (if (or (string= web-mode-cur-language "javascript")
+		  (string= web-mode-cur-language "jsx"))
+	      (unless tern-mode (tern-mode))
+	    (if tern-mode (tern-mode -1))))))
+  (add-hook 'web-mode-hook 'company-mode)
+
+  ;; to get completion data for angularJS
+  (use-package ac-html-angular :defer t)
```
Comment (rakanalh) [inline emacs_templates/emacs/elisp/lang-web.el:69]: This and the one below are `autocomplete` dependent.. and i don't really like mixing up ac & company together. So i would exclude these two packages

**Review by rakanalh — CHANGES_REQUESTED** [review]

File: emacs_templates/emacs/elisp/lang-web.el (hunk 2 above)
Comment (CSRaghunandan) [reply emacs_templates/emacs/elisp/lang-web.el:69]: No no, even though they are auto-complete sources, they are used by `company-web` to provide completion for angular 1.5 or bootstrap (if the user wants to use them, they can enable them with `ac-html-angular` or `ac-html-bootstrap` and they will get company completions)
These are sources, so it wouldn't install `autocomplete.el` package.

**Comment by rakanalh** [comment]
Oh btw, while you're at it... maybe you can squash your commits? that's one more thing to try doing using `magit`

File: README.org (hunk 1 above)
Comment (CSRaghunandan) [reply README.org:110]: Really sorry I introduced a conflict. My bad. I'll make sure this doesn't happen again. I need to get around learning proper git workflow.

File: emacs_templates/emacs/elisp/lang-web.el (hunk 2 above)
Comment (rakanalh) [reply emacs_templates/emacs/elisp/lang-web.el:69]: Yea just checked them out... seems you're right

**Review by rakanalh — APPROVED** [review]

**Pull request merged**